Mlem/App/Views/Shared/InteractionBarView.swift

```diff
@@ -0,0 +1,149 @@
+    @Environment(Palette.self) var palette
+    
+    private let leading: [EnrichedWidget]
+    private let trailing: [EnrichedWidget]
+    private let readouts: [Readout]
+    
+    init(post: any Post1Providing, configuration: PostBarConfiguration) {
+        self.leading = .init(post: post, items: configuration.leading)
+        self.trailing = .init(post: post, items: configuration.trailing)
+        self.readouts = configuration.readouts.map { post.readout(type: $0) }
+    }
+
+    var body: some View {
+        HStack(spacing: AppConstants.standardSpacing) {
+            ForEach(leading, id: \.viewId, content: widgetView)
+            InfoStackView(readouts: readouts, showColor: false)
+                .frame(maxWidth: .infinity, alignment: infoStackAlignment)
+            ForEach(trailing, id: \.viewId, content: widgetView)
+        }
+        .frame(height: AppConstants.barIconSize)
+    }
+    
+    var infoStackAlignment: Alignment {
+        switch (leading.isEmpty, trailing.isEmpty) {
+        case (true, false):
+            .leading
+        case (false, true):
+            .trailing
+        default:
+            .center
+        }
+    }
+    
+    @ViewBuilder
+    private func widgetView(_ widget: EnrichedWidget) -> some View {
+        switch widget {
+        case let .action(action):
+            actionView(action)
+        case let .counter(counter):
+            counterView(counter)
+        }
+    }
+    
+    @ViewBuilder
+    private func counterView(_ counter: Counter) -> some View {
+        HStack {
+            if let leadingAction = counter.leadingAction {
+                actionView(leadingAction)
+            }
+            Text(counter.value?.description ?? "")
+                .monospacedDigit()
+                .contentTransition(.numericText(value: Double(counter.value ?? 0)))
+                .animation(.default, value: counter.value)
+                
+            if let trailingAction = counter.trailingAction {
+                actionView(trailingAction)
+            }
+        }
+    }
+    
+    @ViewBuilder
+    private func actionView(_ action: any Action) -> some View {
+        Button {
+            (action as? BasicAction)?.callback?()
+        } label: {
+            actionLabelView(action)
+        }
+        .accessibilityLabel(action.label)
+        .accessibilityAction(.default) {
+            (action as? BasicAction)?.callback?()
+        }
+        .buttonStyle(EmptyButtonStyle())
+        .disabled({
+            if let action = action as? BasicAction {
+                return action.callback == nil
+            } else {
+                return false
+            }
+        }())
+    }
+    
+    @ViewBuilder
+    private func actionLabelView(_ action: any Action) -> some View {
+        let isOn = ((action as? BasicAction)?.disabled ?? false) ? false : action.isOn
+        Image(systemName: action.barIcon)
+            .resizable()
+            .fontWeight(.medium)
```

Comment:
In v1 the arrows are `.regular` and the save button is `.medium`, which to my eye looks nicer--`.medium` arrows are a little chunky. Can discuss further in Slack.